CONTRIBUTING.md

```diff
@@ -131,89 +131,96 @@ changes.
-* Avoid deep nesting (especially of functions), except where necessary
-  (e.g. due to closure scope).
-* End with a single new-line character.
-* Expose public methods by declaring them on the class's prototype.
-* Within a given function's scope, do not mix declarations and imperative
-  code, and  present these in the following order:
-  * First, variable declarations and initialization.
-  * Second, function declarations.
-  * Third, imperative statements.
-  * Finally, the returned value.
-
+1. Write clean code. Here’s a good summary - https://github.com/ryanmcdermott/clean-code-javascript.
+1. Include JSDoc for any exposed API (e.g. public methods, classes).
+1. Include non-JSDoc comments as-needed for explaining private variables,
+   methods, or algorithms when they are non-obvious. Otherwise code 
+   should be self-documenting.
+1. Classes and Vue components should use camel case, first letter capitalized
+   (e.g. SomeClassName).
+1. Methods, variables, fields, events, and function names should use camelCase,
+   first letter lower-case (e.g. someVariableName).
+1. Source files that export functions should use camelCase, first letter lower-case (eg. testTools.js)
+1. Constants (variables or fields which are meant to be declared and 
+   initialized statically, and never changed) should use only capital 
+   letters, with underscores between words (e.g. SOME_CONSTANT). They should always be declared as `const`s
+1. File names should be the name of the exported class, plus a .js extension
+   (e.g. SomeClassName.js).
+1. Avoid anonymous functions, except when functions are short (one or two lines)
+   and their inclusion makes sense within the flow of the code
+   (e.g. as arguments to a forEach call). Anonymous functions should always be arrow functions.
+1. Named functions are preferred over functions assigned to variables.
+   eg.
+   ```JavaScript
+   function renameObject(object, newName) {
+       Object.name = newName;
+   }
+   ```
+   is preferable to
+   ```JavaScript
+   const rename = (object, newName) => {
+       Object.name = newName;
+   }
+   ```
+1. Avoid deep nesting (especially of functions), except where necessary
+   (e.g. due to closure scope).
+1. End with a single new-line character.
+1. Always use ES6 `Class`es and inheritence rather than the pre-ES6 prototypal 
+   pattern.
+1. Within a given function's scope, do not mix declarations and imperative
+   code, and  present these in the following order:
+   * First, variable declarations and initialization.
+   * Secondly, imperative statements.
+   * Finally, the returned value. Functions should only have a single return statement.
+1. Avoid the use of "magic" values.
+   eg.
+   ```JavaScript
+   Const UNAUTHORIZED = 401
+   if (responseCode === UNAUTHORIZED)
+   ```
+   is preferable to
+   ```JavaScript
+   if (responseCode === 401)
+   ```
+1. Don’t use the ternary operator. Yes it's terse, but there's probably a clearer way of writing it.
```

Comment:
I like `ternary operator` 😭

Reply:
I like it too, but it doesn't like other developers, in my experience 😃 There are rarely cases where a ternary is clearer than an if statement, a ternary usually requires more mental processing.

Ternary operators serve no function other than to make code shorter. When deciding between clarity vs brevity, we should always err on the side of clarity.

Reply:
If absolutely everyone disagrees with me though, I'm happy to reconsider 😄

Reply:
I used to hate it before but I started with following standard mentioned below: 😃 
```
valueX ===  valueY
     ? someValue
     : defaultValue;
```

Reply:
I went in the opposite direction. I used to love it, but I've been persuaded by arguments that it obfuscates code. `if` is human language, and allows you to write clear declarative code. It allows you write clear statements about what the code is doing. `?` and `:` are not human language and result in terse, less readable code.

As I say, if everyone disagrees with me, happy to revise this rule.

Reply:
This is good though, I'm glad we're discussing this. It was suspiciously easy to get a group of developers to agree on a new coding standard!

Reply:
I also like the ternary.

I think only if a non-javascript developer looked at it would it be confusing.

Reply:
>>> I think only if a non-javascript developer looked at it would it be confusing.

I take your point, and agree in the case of simple ternary statements. I regularly see ternary statements being abused though, with nested ternaries, and complex logic which does cause additional confusion. 

Also, it's not just about confusion, it's about reducing the cognitive load of someone trying to understand the code you've written. I think that even in the simplest cases an `if` statement is clearer. The less time our brains have to spend being JavaScript interpreters, the better.

Reply:
hmm, so simple and non nested ternary statements are ok and let reviewer decide if it is too confusing?

Reply:
My argument is that using a ternary statement is always less clear than an if statement, even in the simple case. Write code for humans, not for machines.

@shefalijoshi @deeptailor We need a deciding vote here 😄  If there's a consensus supporting ternaries I'll change the code standard to allow simple ternaries, and disallow nested ones.

Reply:
First of all, sorry for not contributing to this discussion earlier. I agree that clarity is of the upmost importance - especially when dealing with an open source codebase with various authors and preferences. I don't think however that we can objectively say one way or the other which is more clear.

I think the JS guidelines you linked to @akhenry were good in saying when and how to use a ternary and when and how not to use a ternary. That's just me going from memory though...

These are stretches... but are definitely reasons I use them.
- they work in Vue templates where if blocks do not (although most times I favor computeds over template expressions)
- they are exactly what they say they are, whereas `if / else / else if` can take more brain power depending on how they are written

Reply:
I like ternary's when you are trying to assign a variable, 
eg let var = isObjectDefined ? object : someOtherVariable

But excessive use of ternaries can get confusing and mind boggling. My vote it to keep ternary operations to variable assignments only, and to the reviewers decision wether it is confusing or not.

Reply:
cool, I'll change it to allow simple ternaries for things such as defaulting a variable, and disallow nested ternaries.